backpack/core/derivatives/lstm.py

```diff
@@ -0,0 +1,358 @@
+                    + einsum("hg, ng -> nh", module.weight_hh_l0, module.output[t - 1])
+                    + module.bias_hh_l0
+                )
+            ifgo[t, :, H0:H1] = sigmoid(ifgo[t, :, H0:H1])
+            ifgo[t, :, H1:H2] = sigmoid(ifgo[t, :, H1:H2])
+            ifgo[t, :, H2:H3] = tanh(ifgo[t, :, H2:H3])
+            ifgo[t, :, H3:H4] = sigmoid(ifgo[t, :, H3:H4])
+            if t == 0:
+                c[t] = ifgo[t, :, :H] * ifgo[t, :, H2:H3]
+            else:
+                c[t] = (ifgo[t, :, :H] * ifgo[t, :, H2:H3]) + (
+                    ifgo[t, :, H:H2] * c[t - 1]
+                )
+            c_tanh[t] = tanh(c[t])
+            h[t] = ifgo[t, :, H3:] * c_tanh[t]
+
+        # h is the same as previous forward pass
+        assert allclose(h, module.output, atol=1e-5)
+        return ifgo, c, c_tanh, h
+
+    @classmethod
+    def _ifgo_jac_t_mat_prod(cls, module: LSTM, mat: Tensor) -> Tensor:
+        V: int = mat.shape[0]
+        T: int = mat.shape[1]
+        N: int = mat.shape[2]
+        H: int = module.hidden_size
+        H0: int = 0 * H
+        H1: int = 1 * H
+        H2: int = 2 * H
+        H3: int = 3 * H
+        H4: int = 4 * H
+
+        ifgo, c, c_tanh, _ = cls._forward_pass(module, mat)
+
+        # backward pass
+        H_prod: Tensor = zeros(V, T, N, H, device=mat.device)
+        C_prod: Tensor = zeros(V, T, N, H, device=mat.device)
+        IFGO_prod: Tensor = zeros(V, T, N, 4 * H, device=mat.device)
+        for t in range(T)[::-1]:
+            # jac_t_mat_prod until node h
+            if t == T - 1:
+                H_prod[:, t] = mat[:, t]
+            else:
+                H_prod[:, t] = mat[:, t] + einsum(
+                    "vnh, hg -> vng",
+                    IFGO_prod[:, t + 1],
+                    module.weight_hh_l0,
+                )
+
+            # C_prod = jac_t_mat_prod until node c
+            if t == T - 1:
+                C_prod[:, t] = einsum(
+                    "vnh, nh, nh -> vnh",
+                    H_prod[:, t],
+                    ifgo[t, :, H3:H4],
+                    (1 - c_tanh[t] ** 2),
+                )
+            else:
+                C_prod[:, t] = einsum(
+                    "vnh, nh, nh -> vnh",
+                    H_prod[:, t],
+                    ifgo[t, :, H3:H4],
+                    (1 - c_tanh[t] ** 2),
+                ) + einsum(
+                    "vnh, nh -> vnh",
+                    C_prod[:, t + 1],
+                    ifgo[t + 1, :, H1:H2],
+                )
+
+            IFGO_prod[:, t, :, H3:] = einsum(
+                "vnh, nh, nh -> vnh",
+                H_prod[:, t],
+                c_tanh[t],
+                ifgo[t, :, H3:H4] * (1 - ifgo[t, :, H3:H4]),
+            )
+            IFGO_prod[:, t, :, :H] = einsum(
+                "vnh, nh, nh -> vnh",
+                C_prod[:, t],
+                ifgo[t, :, H2:H3],
+                ifgo[t, :, H0:H1] * (1 - ifgo[t, :, H0:H1]),
```

Comment:
This is the derivative of a sigmoid: `sigmoid'(x) = sigmoid(x) * (1 - sigmoid(x))`. It would be great if it were possible to recycle code from the `SigmoidDerivatives`, although I think this is currently not possible due to its interface. Can you open an issue for that on `backpack-discuss`?

Reply:
https://github.com/fKunstner/backpack-discuss/issues/106

Reply:
I've got a chicken-egg problem. In the created issue, how do I reference the lines that could be refactored? (with the issue that these lines are not on f-dangel/backpack yet)

Reply:
Maybe class + method?